bdsbilling/src/main/java/com/appcoins/wallet/bdsbilling/repository/RemoteRepository.kt

```diff
@@ -121,30 +121,36 @@ class RemoteRepository(private val api: BdsApi, private val responseMapper: BdsA
         null).ignoreElement()
   }
 
-  fun createLocalPaymentTopUpTransaction(paymentId: String, packageName: String, price: String,
-                                         currency: String, productName: String,
-                                         walletAddress: String,
-                                         walletSignature: String): Single<Transaction> {
-    return createTransaction(walletAddress, null, null, null, null, null, null, null, null, null,
-        "TOPUP", "myappcoins", walletAddress, walletSignature, packageName, price, currency,
-        null, LocalPaymentBody(price, currency, packageName, "TOPUP", paymentId, productName))
+  fun createLocalPaymentTransaction(paymentId: String, packageName: String, price: String?,
+                                    currency: String?, productName: String?, type: String,
+                                    origin: String?, walletsDeveloper: String?,
+                                    walletsStore: String?, walletsOem: String?,
+                                    developerPayload: String?, callback: String?,
+                                    orderReference: String?, referrerUrl: String?,
+                                    walletAddress: String,
+                                    walletSignature: String): Single<Transaction> {
+    return createTransaction(walletAddress, walletsDeveloper, walletsStore, walletsOem, null,
+        developerPayload, callback, orderReference, referrerUrl, origin, type, "myappcoins",
+        walletAddress, walletSignature, packageName, price, currency, productName,
+        LocalPaymentBody(price, currency, packageName, type, paymentId, productName,
+            walletsDeveloper))
   }
 
   private fun createTransaction(userWallet: String?, developerWallet: String?, storeWallet: String?,
                                 oemWallet: String?, token: String?, developerPayload: String?,
                                 callback: String?, orderReference: String?, referrerUrl: String?,
                                 origin: String?, type: String, gateway: String,
                                 walletAddress: String, signature: String, packageName: String,
-                                amount: String, currency: String, productName: String?,
-                                localPaymentBody: LocalPaymentBody = LocalPaymentBody()): Single<Transaction> {
-    // TODO We should not do this verification by using the payment gateway
-    return if (gateway == "myappcoins") {
-      api.createTransaction(null, packageName, amount, currency, productName,
-          type, walletAddress, null, null, null, null, null, null, null, null, walletAddress,
-          signature, localPaymentBody)
+                                amount: String?, currency: String?,
+                                productName: String?,
+                                localPaymentBody: LocalPaymentBody? = null): Single<Transaction> {
+    return if (localPaymentBody != null) {
```

Comment:
I don't think we need this if clause neither two different method to create the transaction. We can add the gateway in the createLocalPaymentTransaction method and setting the localPaymentBody with default value as null.